roundtripper/roundtripper.go

```diff
@@ -90,22 +90,37 @@ func NewTransport(ctx context.Context, logger *zap.SugaredLogger) http.RoundTrip
 		InnerTransport: transport,
 	}
 
-	// Wrap that with the response cacher
-	cache := &CachingRoundTripper{
-		Logger:         logger,
-		innerTransport: rateLimit,
-		respCache:      map[url.URL]*http.Response{},
-		bodyCache:      map[url.URL][]byte{},
+	// uses the disk cache
+	if cachePath, useDisk := shouldUseDiskCache(); useDisk {
+		c := cache.NewTransport(diskcache.New(cachePath))
+		c.Transport = rateLimit
+		return c
 	}
 
-	return cache
+	// uses memory cache
+	c := cache.NewTransport(cache.NewMemoryCache())
+	c.Transport = rateLimit
+	return c
+}
+
+// shouldUseDiskCache checks the env variables USE_DISK_CACHE and DISK_CACHE_PATH to determine if
+// disk should be used for caching.
+func shouldUseDiskCache() (string, bool) {
+	if isDiskCache := os.Getenv(UseDiskCache); isDiskCache != "" {
```

Comment:
nit: I think you can avoid this if statement and go straight into ParseBool since "" parses as false.